src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/ApiBehaviorConventions/ClientErrorResultFilterConvention.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    /// <summary>
+    /// An <see cref="IActionModelConvention"/> that adds a <see cref="IFilterMetadata"/>
+    /// to <see cref="ActionModel"/> that transforms <see cref="IClientErrorActionResult"/>.
+    /// </summary>
+    public class ClientErrorResultFilterConvention : IActionModelConvention
+    {
+        private readonly ApiBehaviorOptions _apiBehaviorOptions;
+        private readonly ClientErrorResultFilter _clientErrorResultFilter;
+
+        public ClientErrorResultFilterConvention(
+            IOptions<ApiBehaviorOptions> apiBehaviorOptions,
+            IClientErrorFactory clientErrorFactory,
+            ILoggerFactory loggerFactory)
```

Comment:
So anything here that requires services will be hard for users to use with options. I don't have a good solution in mind. If we really wanted to be friendly, we'd refactor of the places that we need services in filters to be factories rather than instances.

Reply:
I mean `IFilterFactory`, not factories for the conventions 😆